src/DynamicNLPModels.jl

```diff
@@ -652,9 +707,9 @@ function _build_H(
     end
 
     ns = size(Q, 1)
-    nr = size(R, 1)
+    nu = size(R, 1)
 
-    H = SparseArrays.sparse([],[],Float64[],(ns * (N + 1) + nr * N), (ns * (N+1) + nr * N))
+    H = SparseArrays.sparse([],[],Float64[],(ns * (N + 1) + nu * N), (ns * (N+1) + nu * N))
```

Comment:
Try to not assume that the float precision is `Float64`. It is better to infer it from one of the function arguments